subsys/testsuite/Kconfig

```diff
@@ -144,4 +144,23 @@ config TEST_ARM_CORTEX_M
 	  Note that by default, when building with ARM_SECURE_FIRMWARE
 	  set, these exceptions are set to target the Non-Secure state.
 
+config NEGATIVE_TEST_FATAL_HANDLER
+	bool "Using a pre-defined fatal handler to deal with negative case"
```

Comment:
Do you really need these to be under a kconfig?  It looks like if you don't call them, none of this code will be linked in.  If that's the case then just make them regular APIs.  You put APIs inside a kconfig if there's non-zero overhead of some form to having them available.

Reply:
Excuse me, I thought they cannot be regular API, because this is only used for some test cases. And put it under kconfig would make it more flexible, for example, if we want to define a totally different k_sys_fatal_handler with this one, we don't need to do anything. Another advantage is adding the code this way won't influence other test cases in our code base.  That's what I thought, how do you think? thank you so much!

Reply:
As I read it, these are just functions, no?  There's no static data with an init hook or anything else that will be automatically linked into the app.  If you don't call the API, then it won't be part of the application.

Reply:
I tried to make them not under Kconfig but the it will show "multiple definition of k_sys_fatal_handler" during compilations, conflicts with other test case such as mem_protect,  pipe, etc, which already define k_sys_fatal_handler. That why I put them under Kconfig. Excuse me, maybe I have something misunderstanding?

Reply:
Ah... OK.  I didn't realize that was a weak symbol and not something registered at runtime.  I guess this is fine, though you could still do this by integrating it with the existing weak fallback version.